teps/0089-spire-support.md

```diff
@@ -0,0 +1,262 @@
+1. Tekton Chains observes the TaskRun, and verifies that SVID and signature against the TaskRun yaml
+1. If verification is successful, Chains proceeds normally. Otherwise, it stops and doesn't sign anything!
+
+
+Then, Chains gets the TaskRun.
+Chains will:
+1. Verify the signature against the SPIRE SVID
+1. If verification fails, Chains will mark the TaskRun as "failed" and move on
+1. Otherwise, continue signing stuff! 
+
+## Implementation Plan
+The plan to achieve non-falsifiable provenance will be implemented in phases.
+
+**Phase 1**
+* Add support for Signed Results with SPIRE (this will primarily involve modifications to the entrypointer image)
+* Add support for Chains verifying Signed Results
+
+**Phase 2**
+* Implement Signed TaskRuns with SPIRE (requires further design)
+* Determine an alternate release process for the SPIRE feature (since the Pipelines Controller will need the SPIRE volume mounted in)
+* Add support for Chains verifying Signed TaskRuns
+
+In parallel with this work, we should:
+* Confirm that this meets the SLSA defintion of "non-falsifiable", which might require a security audit
+
+### Risks and Mitigations
+We'll need to depend on the [github.com/spiffe/go-spiffe](https://github.com/spiffe/go-spiffe) library to interact with the SPIRE agent, request SVIDs and signatures.
+
+
+## Design Details - Signed Results
+
+For now, this design section will focus on Part 1 of the implementation plan: Signed Results.
+This TEP will be updated as we flesh out the design for Signed Taskruns.
+
+### Enabling SPIRE on the Entrypointer Image
+We can add a feature flag `--enable-spire=true` as described in [Customizing the Pipelines Controller behavior](https://github.com/tektoncd/pipeline/blob/main/docs/install.md#customizing-the-pipelines-controller-behavior) as an alpha feature.
+If the feature is enabled, then Pipelines would mount in the `csi` Volume into all Pods.
+
+The entrypointer image should also be able to see that this flag is set and accordingly sign Results.
+
+### Signing Results
+Once results are available to the entrypointer image, it will request a signature and SVID over each Results.
+The signature and SVID will be stored as annotations on the TaskRun, and can be verified by a client.
+
+### Verification in Chains
+Once Signed Results are available, we'll add verification of Signed Results to Chains.
+If verification fails for a TaskRun then Chains will not sign it.
+
+
+## Test Plan
+Tests for Signed Results:
+1. Enabling the alpha feature for SPIRE in Tekton
+1. Requesting an SVID & signature over Results for a TaskRun
+1. Verification of SPIRE with Chains
+1. Verify that a TaskRun that isn't created by Tekton isn't signed Chains
+
+Tests for Signed TaskRuns:
+1. Verify that a TaskRun that's been modified during execution isn't verified
+1. Verify that a TaskRun that's been modified after execution isn't verified
+
+
+
+## Alternatives
+
+### Kubernertes Service Account Token Volume Projection 
```

Comment:
i wonder if we'd want to support this as an option too in the future - do you think there's anything about the current proposal that would preclude this from adding this as an option in the future? (i.e. letting users choose different ways of enforcing non-falsifiabilty)

Reply:
i think they should definitely be able to coexist, nothing in here should prevent that.